src/address.c

```diff
@@ -314,3 +314,80 @@ int verifyP2pkhAddress(char* p2pkh_pubkey, uint8_t len)
     free(dec);
     return true;
 }
+
+/**
+ * @brief This function generates a derived child key from a masterkey using
+ * a custom derived path in string format.
+ * 
+ * @param masterkey The master key from which children are derived from.
+ * @param derived_path The path to derive an address from according to BIP-44.
+ * e.g. m/44h/3h/1/1/1 representing m/44h/3h/account/ischange/index
+ * @param outaddress The derived address.
+ * @param outprivkey The boolean value used to derive either a public or 
+ * private address. 'true' for private, 'false' for public
+ * 
+ * @return 1 if a derived address was successfully generated, 0 otherwise
+ */
+int getDerivedHDAddressByPath(const char* masterkey, const char* derived_path, char* outaddress, bool outprivkey) {
+    if (!masterkey || !derived_path || !outaddress) {
+        debug_print("%s", "missing input\n");
+        return false;
+    }
+
+    /* determine if mainnet or testnet/regtest */
+    const dogecoin_chainparams* chain = chain_from_b58_prefix(masterkey);
+    bool ret = true;
+    dogecoin_hdnode node, nodenew;
+    
+    if (!dogecoin_hdnode_deserialize(masterkey, chain, &node)) {
+        ret = false;
+    }
+    
+    /* derive child key, use pubckd or privckd */
+    if (!dogecoin_hd_generate_key(&nodenew, derived_path, outprivkey ? node.private_key : node.public_key, node.chain_code, !outprivkey)) {
+        ret = false;
+    }
+
+    if (outprivkey) dogecoin_hdnode_serialize_private(&nodenew, chain, outaddress, 200);
+    else dogecoin_hdnode_serialize_public(&nodenew, chain, outaddress, 200);
```

Comment:
I'm not familiar enough with all these different length things. Can this `200` be `112` as well, or will there be other sized results being written to `outaddress`? It just seems odd to have `outaddress[112]` being passed into this function, then down below assuming `outaddress` has a size of `200` even though it really doesn't

Reply:
Have updated to use the HD_MASTERKEY_STRINGLEN constant which is 112.